tests/robot-tests/scripts/create_snapshots.py

```diff
@@ -266,98 +266,133 @@ def write_snapshots_to_file(self) -> None:
+            type=str,
+            required=False,
+        )
+
+        ap.add_argument(
+            "--basic-auth-user",
+            dest="basic_auth_user",
+            help="Username for basic auth",
+            type=str,
+            required=False,
+        )
+
+        ap.add_argument(
+            "--basic-auth-pass",
+            dest="basic_auth_password",
+            help="Password for basic auth",
+            type=str,
+            required=False,
+        )
+
+        group = ap.add_mutually_exclusive_group(required=True)
+        group.add_argument(
+            "--validate",
+            dest="validate",
+            action="store_true",
+            help="Compares the latest snapshots to the existing versions and does not write any snapshot files",
+            required=False,
+        )
+        group.add_argument(
+            "--ci",
+            dest="ci",
+            action="store_true",
+            help="Signifies that the test is running in a CI environment, which will make the test validate snapshots and write any snapshot files which have changed",
+            required=False,
+        )
+
+        ap.add_argument(
+            "--slack-webhook-url",
+            dest="slack_webhook_url",
+            default=None,
+            help="URL for Slack webhook to send notifications when snapshot differences are found during validation",
+            type=str,
+            required=False,
+        )
+
+        ap.add_argument(
+            "-v",
+            "--visual",
+            dest="visual",
+            action="store_true",
+            help="display browser window that the tests run in",
+            required=False,
+        )
+
+        args = ap.parse_args()
+        
+        assert os.path.basename(os.getcwd()) == "robot-tests", "Must run from the robot-tests directory!"
+
+        chrome_options = Options()
+
+        if not args.visual:
+            chrome_options.add_argument("--headless")
+
+        get_webdriver()
+
+        driver = webdriver.Chrome(options=chrome_options)
+
+        if args.basic_auth_user and args.basic_auth_password:
+            driver.execute_cdp_cmd("Network.enable", {})
+            token = base64.b64encode(f"{args.basic_auth_user}:{args.basic_auth_password}".encode())
+            driver.execute_cdp_cmd("Network.setExtraHTTPHeaders", {"headers": {"Authorization": f"Basic {token.decode()}"}})
+
+        snapshot_service = SnapshotService(public_url=args.public_url, slack_webhook_url=args.slack_webhook_url)
+
+        if args.ci:
+            snapshot_service.validate_snapshots()
+            snapshot_service.write_snapshots_to_file()
+
+        if args.validate:
+            snapshot_service.validate_snapshots()
+
+        driver.quit()
+    except Exception as ex:
+        message = ""
+        
+        if hasattr(ex, 'message'):
+            message = ex.message
+        elif hasattr(ex, 'msg'):
+            message = ex.msg
+        else:
+            message = ex
```

Comment:
Wondering where you got this from? I can see people complaining about the message either being `ex.message` or just `ex`, but couldn't find a mention about `ex.msg`.

Reply:
From what I gathered, `message` was the most common property of `Exception`-type classes, and those which derive from them. Unfortunately there doesn't seem to be anything definitive as there's no interface/implementation contract like something a .NET class might have (e.g. if it derives from `Exception`, it _must_ have a `message` property).

`msg` is the equivalent property for Selenium's `NoSuchElementException` - again, there's nothing forcing them to use `message`. It's essentially a free-for-all what the error summary property is called, which is why I added a catch-all `ex` for exceptions which don't have either of those two properties. The problem here is that the full `ex` object is likely to include stack trace information, but other conditions can easily be added to this block in the future if necessary.